internal/pkg/util/nodeutils.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+
+	statusv1alpha1 "sigs.k8s.io/security-profiles-operator/api/secprofnodestatus/v1alpha1"
+)
+
+func GetDynamicClient() (dynamic.Interface, error) {
+	clusterConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("get in-cluster config: %w", err)
+	}
+
+	// Create a dynamic client for working with nodes
+	dynamicClient, err := dynamic.NewForConfig(clusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("load dynamic client: %w", err)
+	}
+	return dynamicClient, nil
+}
+
+func GetNodeList(ctx context.Context) ([]string, error) {
+	dynamicClient, err := GetDynamicClient()
+	if err != nil {
+		return nil, err
+	}
+	// Specify the resource (nodes) and namespace
+	nodeResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "nodes"}
+
+	// List the nodes (using the dynamic client)
+	nodeList, err := dynamicClient.Resource(nodeResource).Namespace("").List(ctx, metav1.ListOptions{})
```

Comment:
How does this call perform for big clusters with hundreds of nodes? Do we need to use any limit here when fetching the node objects?

Reply:
@ccojocar The largest cluster I've got access to is about 150 nodes (although during peak it could grow bigger). I have no access to multi-hundreds of nodes clusters in a test infrastructure. My test cluster is about 50~ nodes. Since I'm trying to determine if all nodes match the finalizers on an object (not just in number) it might be better to skip the check if we see performance issues on multi-hundred node clusters. 

Off the top of my head I can't think of an example of doing a check like this in a performant way to gain some inspiration, do you know of any such check done by another project?